HLTrigger/special/plugins/HLTMuonRecHitClusterFilter.cc

```diff
@@ -89,13 +117,27 @@ bool HLTMuonRecHitClusterFilter::filter(edm::StreamID, edm::Event& iEvent, const
   auto const& rechitClusters = iEvent.get(cluster_token_);
 
   for (auto const& cluster : rechitClusters) {
-    if ((cluster.size() >= min_Size_) && ((cluster.size() - cluster.nMB1()) >= min_SizeMinusMB1_) &&
-        (cluster.nMB1() <= max_nMB1_) && (cluster.nMB2() <= max_nMB2_) && (cluster.nME11() <= max_nME11_) &&
-        (cluster.nME12() <= max_nME12_) && (cluster.nME41() <= max_nME41_) && (cluster.nME42() <= max_nME42_) &&
-        (cluster.nStation() >= min_Nstation_) && (cluster.avgStation() >= min_AvgStation_) &&
-        ((min_Eta_ < 0.0) || (std::abs(cluster.eta()) >= min_Eta_)) &&
-        ((max_Eta_ < 0.0) || (std::abs(cluster.eta()) <= max_Eta_)) && (cluster.time() > min_Time_) &&
-        (cluster.time() <= max_Time_) && (cluster.timeSpread() <= max_TimeSpread_)) {
+    auto passSizeCut = (cluster.size() >= min_Size_ && min_Size_ > 0) ||
+                       ((cluster.size() - cluster.nMB1()) >= min_SizeMinusMB1_ && min_SizeMinusMB1_ > 0);
```

Comment:
```suggestion
    auto passSizeCut = (min_Size_ < 0 || cluster.size() >= min_Size_) ||
                       (min_SizeMinusMB1_ < 0 || (cluster.size() - cluster.nMB1()) >= min_SizeMinusMB1_);
```

Reply:
@missirol Since the filter requires passing any of one the `passSizeCut` (simple size, size-MB1, or SizeRegionCut) , we want the positive logic to activate the cut we want (hence I used `&& min_Size_>0` etc ), because we do not want the cluster passes the filter due to simple size cut if we're setting `SizeRegionCut` only.

Reply:
@kakwok Okay. Sorry for misunderstanding. So, this here is the OR of two conditions, and the implied convention is that, if the size cut-value is negative, the cut is disabled (making the OR stricter). I think this is okay; it would arguably be more readable to have a separate flag (e.g. `applyMinSizeMinusMB1Cut = True`) in the PSet, but it's clear by now that to really know the logic of the filter one has to look inside of it anyway.

Reply:
@missirol Right, I think this is a small price to pay when we want to cover as much use case as possible with one filter.